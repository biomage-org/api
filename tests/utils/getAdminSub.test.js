const getAdminSub = require('../../src/utils/getAdminSub');
const config = require('../../src/config');

jest.mock('../../src/api.v2/helpers/cognito/getAwsPoolId');
jest.mock('../../src/config', () => ({
  cognitoISP: {
    adminGetUser: jest.fn(() => ({
      promise: () => Promise.resolve({
        Username: 'mockAdminSub',
      }),
    })),
    adminCreateUser: jest.fn(() => ({
      promise: () => Promise.resolve({
        User: {
          Username: 'newMockAdminSub',
        },
      }),
    })),
  },
}));

describe('getAdminSub', () => {
  it('Should return the correct admin sub', async () => {
    const adminSub = await getAdminSub();

    expect(config.cognitoISP.adminGetUser.mock.calls[0]).toMatchSnapshot();
    expect(adminSub).toEqual('mockAdminSub');
  });

  it('Should create a new admin user if admin user does not exist', async () => {
    config.cognitoISP.adminGetUser.mockImplementationOnce(() => ({
      promise: () => Promise.reject(new Error('User does not exist')),
    }));
    const adminSub = await getAdminSub();

    expect(adminSub).toEqual('newMockAdminSub');
    expect(config.cognitoISP.adminCreateUser.mock.calls[0]).toMatchSnapshot();
  });

  it('Should throw an error if there are other errors', async () => {
    config.cognitoISP.adminGetUser.mockImplementationOnce(() => ({
      promise: () => Promise.reject(new Error('Some other error')),
    }));

    expect(async () => {
      await getAdminSub();
    }).rejects.toThrow();
  });
});
