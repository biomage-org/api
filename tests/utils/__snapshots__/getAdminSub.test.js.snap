// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAdminSub Should create a new admin user if admin user does not exist 1`] = `
Array [
  Object {
    "MessageAction": "SUPPRESS",
    "UserAttributes": Array [
      Object {
        "Name": "email",
        "Value": "cellenics_admin@listserv.med.harvard.edu",
      },
      Object {
        "Name": "name",
        "Value": "Biomage Admin",
      },
      Object {
        "Name": "email_verified",
        "Value": "true",
      },
    ],
    "UserPoolId": "trivial-pool-id",
    "Username": "cellenics_admin@listserv.med.harvard.edu",
  },
]
`;

exports[`getAdminSub Should return the correct admin sub 1`] = `
Array [
  Object {
    "UserPoolId": "trivial-pool-id",
    "Username": "cellenics_admin@listserv.med.harvard.edu",
  },
]
`;
